src/content/plugins/module-concatenation-plugin.md

```diff
@@ -3,16 +3,20 @@ title: ModuleConcatenationPlugin
 contributors:
   - skipjack
   - TheLarkInn
+  - lukastong
+
 related:
   - webpack 3: Official Release!!
 ---
 
-In the past, one of webpack’s trade-offs when bundling was that each module in your bundle would be wrapped in individual function closures. These wrapper functions made it slower for your JavaScript to execute in the browser. In comparison, tools like Closure Compiler and RollupJS ‘hoist’ or concatenate the scope of all your modules into one closure and allow for your code to have a faster execution time in the browser.
 
-This plugin will enable the same concatenation behavior in webpack.
+过去 `webpack` 打包时的一个取舍是将 `bundle` 中各个模块单独打包成闭包。这些打包函数使你的 `Javascript` 在浏览器中处理的更慢。相比之下，一些工具像 `Closure Compiler` 和 `RollupJs` 可以提升 ( hoist ) 或者预编译所有模块到一个闭包中，提升你的代码在浏览器中的执行速度。
+
+
+这个插件会在 `webpack` 中实现以上的预编译功能
 
 ``` js
 new webpack.optimize.ModuleConcatenationPlugin()
 ```
 
-> Scope Hoisting is specifically a feature made possible by ECMAScript Module syntax. Because of this webpack may fallback to normal bundling based on what kind of modules you are using, and [other conditions](https://medium.com/webpack/webpack-freelancing-log-book-week-5-7-4764be3266f5).
+> `Scope Hoisting` 是 `ECMAScrip` 模块语法的特性，此特性让 `webpack` 仅打包你用得到的模型到 `bundle` 中和 [其他的情况 ](https://medium.com/webpack/webpack-freelancing-log-book-week-5-7-4764be3266f5).
```

Comment:
> 此特性让 `webpack` 仅打包你用得到的模型到 `bundle` 中和 [其他的情况 ](https://medium.com/webpack/webpack-freelancing-log-book-week-5-7-4764be3266f5).

这一句翻译的意思好像有点问题。

Reply:
是有点别扭...我改下

Reply:
@Aladdin-ADD `Scope Hoisting` 是 `ECMAScript` 语法模块的特性，因此特性 `webpack` 会退回到仅包含你正在使用的各类模块的 `normal bundling`  你看这么翻呢？

Reply:
made possible by ECMAScript Module syntax.
因为 ES6 模块才成为可能。

Reply:
> Because of this webpack may fallback to normal bundling based on what kind of modules you are using, and [other conditions]

webpack可能会根据你正在使用的模块和其他情况，回退到normal bundling。

Reply:
这个 normal 应该对应不同类型模块的 normal 吧， cjs, AMD, CMD啥的?

Reply:
要是说 "回退到普通bundling" 觉得怪怪的.